File: src/main/java/com/curie/curieapp/entities/Meta.java
package com.curie.curieapp.entities;

import com.curie.curieapp.entities.enums.Prioridade;
import com.curie.curieapp.entities.enums.Status;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "metas")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Meta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String objetivo;

    @Column(nullable = false)
    private String descricao;

    @Column(nullable = false)
    private LocalDateTime inicio;

    @Column(nullable = false)
    private LocalDateTime fim;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Prioridade prioridade;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;
}

File: app/src/main/java/com/curie/curie/data/model/Meta.kt
package com.curie.curie.data.model

import com.curie.curie.data.model.enums.Prioridade
import com.curie.curie.data.model.enums.Status

data class Meta (
    val id: Long,
    val userId: Long,
    val objetivo: String,
    val descricao: String,
    val inicio: String,
    val fim: String,
    val prioridade: Prioridade,
    val status: Status
)